src/Catalyst.Core.Modules.Dfs/BlockExchange/Bitswap11.cs

```diff
@@ -0,0 +1,238 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+using Lib.P2P;
+using MultiFormats;
+using ProtoBuf;
+using Semver;
+using ProtoBufHelper = Lib.P2P.ProtoBufHelper;
+
+#pragma warning disable 0649 // disable warning about unassinged fields
+#pragma warning disable 0169// disable warning about unassinged fields
+
+namespace Catalyst.Core.Modules.Dfs.BlockExchange
+{
+    /// <summary>
+    ///   Bitswap Protocol version 1.1.0 
+    /// </summary>
+    public class Bitswap11 : IBitswapProtocol
+    {
+        private static ILog log = LogManager.GetLogger(typeof(Bitswap11));
+
+        /// <inheritdoc />
+        public string Name { get; } = "ipfs/bitswap";
+
+        /// <inheritdoc />
+        public SemVersion Version { get; } = new SemVersion(1, 1);
+
+        /// <inheritdoc />
+        public override string ToString() { return $"/{Name}/{Version}"; }
+
+        /// <summary>
+        ///   The <see cref="Bitswap"/> service.
+        /// </summary>
+        public Bitswap Bitswap { get; set; }
+
+        /// <inheritdoc />
+        public async Task ProcessMessageAsync(PeerConnection connection,
+            Stream stream,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            // There is a race condition between getting the remote identity and
+            // the remote sending the first wantlist.
+            await connection.IdentityEstablished.Task.ConfigureAwait(false);
+
+            while (true)
+            {
+                var request = await ProtoBufHelper.ReadMessageAsync<Message>(stream, cancel).ConfigureAwait(false);
+
+                // Process want list
+                if (request.wantlist != null && request.wantlist.entries != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)